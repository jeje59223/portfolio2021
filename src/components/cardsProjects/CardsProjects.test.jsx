import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import CardsProjects from './CardsProjets';

configure({ adapter: new Adapter() });

describe('<CardsProjects>', () => {
    const firstProject = {
            "id": 1,
            "title": "Games-Catalog",
            "description": "Site qui présente des consoles et des jeux vidéo, avec une partie admin pour la partie gestion des consoles et des jeux. Ce site a été réalisé en NextJS, Bootstrap et du CSS",
            "image": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQfHAOj4KbzZVRAZtRhdo1hG-DnZJYJ_OW8bw&usqp=CAU",
            "technologies": [
                {
                    "name": "NextJS"
                },
                {
                    "name": "Bootstrap"
                },
                {
                    "name": "CSS"
                }
            ]
        }

    it('should render CardsProjects', () => {
        const wrapper = shallow(<CardsProjects projet={firstProject} />);
        const json = toJson(wrapper);
        expect(json).toMatchSnapshot();
    });
});