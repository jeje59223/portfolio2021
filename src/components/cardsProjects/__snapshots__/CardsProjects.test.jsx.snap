// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsProjects> should render CardsProjects 1`] = `
<div
  className="col-12 col-md-6 col-lg-4"
  key="1"
>
  <Card
    body={false}
    className="projetsCards m-5"
    style={
      Object {
        "minWidth": "18rem",
      }
    }
  >
    <CardHeader
      className="bg-dark cardhead"
    >
      <CardImg
        alt="Games-Catalog"
        className="img"
        src="dbz.png"
        variant="top"
      />
    </CardHeader>
    <CardBody
      className="text-center cardBody"
    >
      <CardTitle>
        Games-Catalog
      </CardTitle>
      <CardText>
        Site qui présente des consoles et des jeux vidéo, avec une partie admin pour la partie gestion des consoles et des jeux. Ce site a été réalisé en NextJS, Bootstrap et du CSS
      </CardText>
    </CardBody>
    <CardFooter
      className="text-muted text-center"
    >
      <Button
        active={false}
        disabled={false}
        variant="secondary"
      >
        Details
      </Button>
    </CardFooter>
  </Card>
</div>
`;
